CorsixTH/Src/th_gfx_sdl.cpp

```diff
@@ -397,17 +397,26 @@ bool THRenderTarget::fillRect(uint32_t iColour, int iX, int iY, int iW, int iH)
 void THRenderTarget::getClipRect(THClipRect* pRect) const
 {
     SDL_RenderGetClipRect(m_pRenderer, reinterpret_cast<SDL_Rect*>(pRect));
+    if (pRect->w == 0 || pRect->h == 0) {
```

Comment:
I would create a temp `pSDLRect`, pass that to `SDL_RenderGetClipRect`, and then check  `SDL_RectEmpty(pSDLRect)`. That way you're consistent with `THRenderTarget::setClipRect`. It should be exactly the same check under the hood, though.